rusk-wallet/src/rues.rs

```diff
@@ -49,8 +49,10 @@ impl RuesHttpClient {
     }
 
     /// Check rusk connection
-    pub async fn check_connection(&self) -> Result<(), reqwest::Error> {
-        reqwest::Client::new().post(&self.uri).send().await?;
+    pub async fn check_connection(&self) -> Result<(), Error> {
+        let graphql = GraphQL::new(self.uri.to_owned(), |_: &str| {});
+        graphql.check_connection().await?;
+
```

Comment:
yes we need to check properly if the network is 404 or not because we tell the user if it is in the main menu

Reply:
> yes we need to check properly if the network is 404 or not because we tell the user if it is in the main menu

I don't like that this was put in here, because it currently pollutes other behavior we have. For example, if I try to unshield funds while not having a shielded balance, I get the error related to having a too low balance, but it gets overridden by the 404 error.

Reply:
Bought back old check (which is broken)